vortex-array/src/array/struct_/mod.rs

```diff
@@ -143,9 +143,11 @@ impl ArrayVariants for StructArray {
 
 impl StructArrayTrait for StructArray {
     fn field(&self, idx: usize) -> Option<Array> {
-        self.dtypes()
-            .get(idx)
-            .and_then(|dtype| self.as_ref().child(idx, dtype, self.len()))
+        self.dtypes().get(idx).map(|dtype| {
+            self.as_ref()
+                .child(idx, dtype, self.len())
+                .unwrap_or_else(|e| vortex_panic!(e, "StructArray: field {} not found", idx))
```

Comment:
Weird for this to panic instead of returning None

Reply:
I didn't want to go from Result -> Option b/c then you silently drop the error information.

Arrow struct array equivalently panics (it translates into a Vec out-of-bounds access).

Tbh I'm not sure why this trait returns Option<Array>, I'm not sure what a caller is going to do with the None

Reply:
I think this is the case where you have malformed array since dtypes and arrays don't match

Reply:
But for a malformed array you probably want the error and not the None. Maybe the trait method should be returning VortexResult<Array> ?

Reply:
Let’s not go crazy here. You validate on construction. Then here you can panic imho since that’s a bug in structarray construction code

Reply:
I agree with that

Reply:
Sgtm